pkg/storage/sst.go

```diff
@@ -460,12 +463,25 @@ func CheckSSTConflicts(
 					} else {
 						// Add the sst range key versions again, to account for the overlap
 						// with extRangeKeys.
-						for _, v := range sstRangeKeys.Versions {
-							statsDiff.Add(updateStatsOnRangeKeyPutVersion(extRangeKeys, v))
+						updatedStack := extRangeKeys
+						updatedStack.Versions = extRangeKeys.Versions.Clone()
+						for i, v := range sstRangeKeys.Versions {
+							if i == 0 {
+								// We do this dance to make updatedStack.Versions.Newest() == v. This
+								// is necessary to keep GCBytesAge calculations correct, we don't
+								// want updateStatsOnRangeKeyPutVersion to "lift" the GCBytesAge
+								// contribution of extRangeKeys' bounds. We will do that later.
+								// We only want it to add the version.
+								oldVersions := updatedStack.Versions
+								updatedStack.Versions = append(MVCCRangeKeyVersions{}, v)
```

Comment:
nit: Can use `MVCCRangeKeyVersions{v}, oldVersions...)` to use a single allocation here. The dance is also a bit unfortunate, we can consider either inlining the logic or adding a new helper function instead.